go/tools/builders/stdliblist_test.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+func Test_stdliblist(t *testing.T) {
+	testDir := t.TempDir()
+	f, _ := ioutil.TempFile(testDir, "*")
+
+	// test files are at run file directory, but this test is run at
+	// {runfile directory}/bazel.TestWorkspace()
+	// since -sdk is assumed to be a relative path to execRoot
+	// (go.sdk.root_file.dirname), thus setting wd to
```

Comment:
Can make`-sdk` to take either relative path to current directory (execRoot in bazel action) or an absolute path?

Reply:
`-sdk` is passed as relative path in stdlib.bazl, and stdliblist.go has this assumption too.
Also it looks like we can't get absolute path easily #2713

https://github.com/xytan0056/rules_go/blob/8b0dde29fb0dd2a7a7b35d6e06a43093f8716bc2/go/private/actions/stdlib.bzl#L55-L58

<img width="766" alt="image" src="https://user-images.githubusercontent.com/48696694/168700144-98e3ac17-c9b4-48e5-bc84-669b3e0b75d7.png">

Reply:
Actually -sdk needs to be a relative path, because during cloning, we need to **retain the same relative path of the original go_sdk**, i.e. `/old/root/`**`rel_go_sdk`**`/src/package` -> `/new/root/`**`rel_go_sdk`**`/src/package`

It cannot be an absolute path, because if so, the only way to get `rel_go_sdk` from goroot is `filepath.Rel(".", env.go_sdk)`, but this doesn't work for tests:
consider the following test file structure
```
/a/b/c/runfiles
/a/b/c/runfiles/go_sdk <- data files
/a/b/c/runfiles/workspace <- where test is run with rundir being "." in go_test rule
```
the relative path of go_sdk is `../go_sdk`;
Let's make `cloneBase := /tmp/123`;
Then in order to retain the same relative path ,`newGoRoot`  becomes `/tmp/123/../go_sdk`, which is `/tmp/go_sdk`;
Then `go list` under `newGoRoot` returns `pkg.Dir` being `/tmp/go_sdk/src/package`, which does not share a prefix with `cloneBase` and thus cannot be replaced with `__BAZEL_OUTBASE_LABEL__`
```
func labelledPath(cloneBase, p string) string {
        // ../go_sdk/src/package 
	dir, _ := filepath.Rel(cloneBase, p) 

        // __BAZEL_OUTPUT_BASE__/../go_sdk/src/package => go_sdk/src/package (wrong)
	return filepath.Join("__BAZEL_OUTPUT_BASE__", dir)  
}
```

The culprit here is `..` in the relative path. Instead of maniplulating the path, a simple way is to set `rundir=".."` in the go_test rule, so that all assumptions naturally fall into places.

Reply:
Why not just cloning from `/old/root/external/go_sdk` to `/new/root/external/go_sdk`?

Reply:
Never mind. https://github.com/xytan0056/rules_go/pull/1